Sources/Tentacle/Content.swift

```diff
@@ -57,6 +54,43 @@ public enum Content {
             /// when they are the result of a query for a directory
             /// See https://developer.github.com/v3/repos/contents/
             case submodule(url: String?)
+
+            public init(from decoder: Decoder) throws {
+                let container = try decoder.container(keyedBy: CodingKeys.self)
+                let type = try container.decode(String.self, forKey: .type)
+                switch type {
+                case "file":
+                    let size = try container.decode(Int.self, forKey: .size)
+                    if let url = try container.decodeIfPresent(URL.self, forKey: .downloadURL) {
+                        self = .file(size: size, downloadURL: url)
+                    } else {
+                        self = .submodule(url: nil)
+                    }
+                case "dir":
+                    self = .directory
+                case "submodule":
+                    let url = try container.decodeIfPresent(String.self, forKey: .submoduleURL)
+                    self = .submodule(url: url)
+                case "symlink":
+                    let target = try container.decodeIfPresent(String.self, forKey: .target)
+                    let url = try container.decodeIfPresent(URL.self, forKey: .downloadURL)
+                    self = .symlink(target: target, downloadURL: url)
+                default:
+                    throw DecodingError.dataCorruptedError(
+                        forKey: CodingKeys.type,
+                        in: container,
+                        debugDescription: "Invalid content-type \(type)"
+                    )
+                }
+            }
```

Comment:
🙈

Reply:
How else would you handle this? 🤔

Reply:
Made a proposal in 261c768, let me know what you think. At least we're no longer switching over strings.